src/main/java/com/microsoft/sqlserver/jdbc/SQLServerSecurityUtility.java

```diff
@@ -319,191 +319,109 @@ static void verifyColumnMasterKeyMetadata(SQLServerConnection connection, SQLSer
-            identityEndpoint = System.getenv("MSI_ENDPOINT");
-        }
+        Optional<AccessToken> accessTokenOptional = mic
+                .getToken(tokenRequestContext)
+                .blockOptional();
 
-        String identityHeader = System.getenv("IDENTITY_HEADER");
-        if (null == identityHeader || identityHeader.trim().isEmpty()) {
-            identityHeader = System.getenv("MSI_SECRET");
-        }
+        if (!accessTokenOptional.isPresent()) {
+            if (connectionlogger.isLoggable(java.util.logging.Level.FINE)) {
+                connectionlogger.fine("Access token is not present.");
+            }
 
-        /*
-         * isAzureFunction is used for identifying if the current client application is running in a Virtual Machine
-         * (without Managed Identity environment variables) or App Service/Function (with Managed Identity environment
-         * variables) as the APIs to be called for acquiring MSI Token are different for both cases.
-         */
-        boolean isAzureFunction = null != identityEndpoint && !identityEndpoint.isEmpty() && null != identityHeader
-                && !identityHeader.isEmpty();
+            throw new SQLServerException(SQLServerException
+                    .getErrString("R_ManagedIdentityTokenAcquisitionFail"), null);
 
-        if (isAzureFunction) {
-            urlString.append(identityEndpoint).append("?api-version=2019-08-01&resource=").append(resource);
         } else {
-            urlString.append(ActiveDirectoryAuthentication.AZURE_REST_MSI_URL).append("&resource=").append(resource);
-            // Retry acquiring access token up to 20 times due to possible IMDS upgrade (Applies to VM only)
-            maxRetry = 20;
-            // Simplified variant of Exponential BackOff
-            for (int x = 0; x < maxRetry; x++) {
-                retrySlots.add(INTERNAL_SERVER_ERROR * ((2 << x) - 1) / 1000);
-            }
+            AccessToken accessToken = accessTokenOptional.get();
+            sqlFedAuthToken = new SqlFedAuthToken(accessToken.getToken(), accessToken.getExpiresAt().toEpochSecond());
         }
 
-        // Append Client Id if available
-        if (null != msiClientId && !msiClientId.isEmpty()) {
-            urlString.append("&client_id=").append(msiClientId);
-        }
+        return sqlFedAuthToken;
+    }
 
-        // Loop while maxRetry reaches its limit
-        while (retry <= maxRetry) {
-            HttpURLConnection connection = null;
+    /**
+     * Get Managed Identity Authentication token through the DefaultAzureCredential
+     *
+     * @param resource
+     *        token resource
+     * @param msiClientId
+     *        Managed Identity or User Assigned Managed Identity
+     * @return fedauth token
+     * @throws SQLServerException
+     */
+    static SqlFedAuthToken getDefaultAzureCredAuthToken(String resource, String msiClientId) throws SQLServerException {

```

Comment:
lets not use the terminalogy "msi" anymore since it's confusing.  Use new terminology "managedIdentity" or "mi" everywhere for consistency